src/Microsoft.Framework.DesignTimeHost/PluginHandler.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.Framework.DesignTimeHost.Models.IncomingMessages;
+using Microsoft.Framework.Runtime;
+using Microsoft.Framework.Runtime.Common.DependencyInjection;
+
+namespace Microsoft.Framework.DesignTimeHost
+{
+    public class PluginHandler
+    {
+        private readonly Action<object> _sendMessageMethod;
+        private readonly IServiceProvider _hostServices;
+        private readonly IDictionary<string, IPlugin> _plugins;
+
+        public PluginHandler(IServiceProvider hostServices, Action<object> sendMessageMethod)
+        {
+            _sendMessageMethod = sendMessageMethod;
+            _hostServices = hostServices;
+            _plugins = new Dictionary<string, IPlugin>(StringComparer.Ordinal);
```

Comment:
if an `ApplicationContext` may be shared between threads, probably need a `ConcurrentDictionary` here